src/ap/java/org/spongepowered/plugin/processor/PluginElement.java

```diff
@@ -181,6 +185,16 @@ void apply(Messager messager) {
                 this.metadata.loadAfter(new PluginMetadata.Dependency(id, dependency.version()), !dependency.optional());
             }
         }
+
+        if (!hasSpongeDependency) {
+            String version = Platform.API_VERSION.orElseThrow(
+                    () -> new IllegalStateException("Can't find SpongeAPI version"));
+            if (version.contains("-")) version = version.substring(0, version.indexOf('-'));
```

Comment:
Example:
```java
Pattern versionPattern = Pattern.compile(".*-?((\\d+)\\.(\\d+\\.)*\\d)");
Matcher versionMatcher = versionPattern.matcher(version);
if (versionMatcher.find()) {
    // This is now safe because group is guaranteed to be digits
    int major = Integer.parseInt(versionMatcher.group(2));
    String range = String.format("[%s,%d.0)", versionMatcher.group(1), ++major);
    this.metadata.loadAfter(....
}
```
Do not use this verbatim, ensure that you understand it. Consult [this site](http://www.regular-expressions.info/) until you understand regular expressions enough to know what a character class and a capturing subpattern is.

Also note the use of `String.format` instead of messy string concatenation for clarity.